bundles/http/Citrix/Slack/2e8a194d-0c01-44c1-9a62-a5ee1d5ecbb3.js

```diff
@@ -0,0 +1,276 @@
+                        } else {
+                            parsedMessages[indexMessages].text_changed = text
+                        }
+
+                        searchText = searchText.substring(searchText.indexOf('>') + 1)
+
+                        let resp = await client.fetch(ENDPOINT_USER_INFO + user_ids[indexUser])
+
+                        if (resp.ok) {
+                            let userData = await resp.json()
+                            if (userData.user === undefined) {
+                                indexChannels++
+                                continue
+                            }
+
+                            parsedMessages[indexMessages].text_changed = text.replace('<@' + user_ids[indexUser] + '>', userData.user.real_name)
+
+                        } else {
+                            throw new Error(`Could not retrieve user data (${response.status}: ${response.statusText})`);
+                        }
+
+                        indexUser++
+                    } while (searchText.indexOf('>') >= 0)
+
+                    parsedMessages[indexMessages].datetime = new Date(parsedMessages[indexMessages].ts * 1000);
+                    parsedChannelMessages.push(parsedMessages[indexMessages])
+
+                    indexMessages++;
+                } while (parsedMessages[indexMessages] !== undefined)
+
+
+            } else {
+                throw new Error(`Could not retrieve favorite channel messages (${response.status}: ${response.statusText})`);
+            }
+
+            indexChannels++;
+        } while (parsedFavoriteChannels[indexChannels] !== undefined)
+
+        urlEndpoint = ENDPOINT_CHANNEL;
+
+        let respChannels = await client.fetch(urlEndpoint)
+
+        if (respChannels.ok) {
+            let allChannels = await respChannels.json()
+            let prettyChannels = JSON.stringify(allChannels.channels, null, '\t')
+            let parsedChannels = JSON.parse(prettyChannels)
+            let indexChannels = 0;
+
+            do {
+                let channel = parsedChannels[indexChannels]
+                urlEndpoint = ENDPOINT_CHANNEL_MEMBERS + channel.id
+                let respChannelsMembers = await client.fetch(urlEndpoint)
+
+                if (respChannelsMembers.ok) {
+                    let allChannelsMember = await respChannelsMembers.json()
+                    if (allChannelsMember.members === undefined) {
+                        indexChannels++
+                        continue
+                    }
+                    let prettyChannelsMembers = JSON.stringify(allChannelsMember.members, null, '\t')
+                    let parsedChannelsMembers = JSON.parse(prettyChannelsMembers)
+                    let insert = true;
+                    let indexChannelsMembers = 0;
+                    let channelMember = "";
+
+                    do {
+
+                        channelMember = parsedChannelsMembers[indexChannelsMembers]
```

Comment:
Why do you declare it before `do` cycle when it is not used outside of `do/while`?

```suggestion
                        const channelMember = parsedChannelsMembers[indexChannelsMembers]
```